From 8f9309f12100c114e85a9004ce51d75e382328c4 Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm.code@gmail.com>
Date: Fri, 15 Sep 2017 19:53:02 +0000
Subject: [PATCH 2/4] e2fsprogs: Build static libs for recovery

 * Add libe2fsck, libmke2fs.

 * Allow libext2_profile to build statically.

Change-Id: If676329230db0dd868453c406392f584e3a24dbe
---
 e2fsck/Android.bp      |  7 +++++++
 lib/support/Android.bp |  2 +-
 misc/Android.bp        | 12 ++++++++++++
 3 files changed, 20 insertions(+), 1 deletion(-)

diff --git a/e2fsck/Android.bp b/e2fsck/Android.bp
index 475cea12..ddfe958f 100644
--- a/e2fsck/Android.bp
+++ b/e2fsck/Android.bp
@@ -66,3 +66,10 @@ cc_binary {
 
     static_libs: e2fsck_libs,
 }
+
+cc_library_static {
+    name: "libe2fsck",
+    defaults: ["e2fsck-defaults"],
+    cflags: ["-Dmain=e2fsck_main"],
+    header_libs: ["libext2-headers"],
+}
diff --git a/lib/support/Android.bp b/lib/support/Android.bp
index 0ddd0793..c5e12569 100644
--- a/lib/support/Android.bp
+++ b/lib/support/Android.bp
@@ -33,7 +33,7 @@ cc_library {
     export_header_lib_headers: ["libext2-headers"],
 }
 
-cc_library_shared {
+cc_library {
     name: "libext2_profile",
     host_supported: true,
     unique_host_soname: true,
diff --git a/misc/Android.bp b/misc/Android.bp
index e474c2d0..8881ebf2 100644
--- a/misc/Android.bp
+++ b/misc/Android.bp
@@ -99,6 +99,18 @@ cc_binary {
     include_dirs: ["external/e2fsprogs/e2fsck"],
 }
 
+cc_library_static {
+    name: "libmke2fs",
+    srcs: [
+        "mke2fs.c",
+        "mk_hugefiles.c",
+        "default_profile.c",
+        "create_inode.c",
+    ],
+    cflags: ["-W", "-Wall", "-Wno-macro-redefined", "-DBUILD_AS_LIB"],
+    include_dirs: ["external/e2fsprogs/lib", "external/e2fsprogs/e2fsck"],
+}
+
 //##########################################################################
 // Build tune2fs
 
-- 
2.17.1

